src/main/java/HeadersRestTemplateService.java

```diff
@@ -0,0 +1,28 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Service;
+
+@Service
+public class HeadersRestTemplateService {
+
+    @Value("${github.token}")
+    private String githubToken;
+
+    @Value("${openai.api.key}")
+    private String openaiApiKey;
+
+    public HttpHeaders createGitHubHeaders() {
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + githubToken);
+        headers.set("Content-Type", "application/json");
+        return headers;
+    }
+
+    public HttpHeaders createOpenAiHeaders() {
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + openaiApiKey);
+        headers.set("Content-Type", "application/json");
+        return headers;
+    }
```

Comment:
Both methods have similar logic, which leads to code duplication. This violates the DRY (Don't Repeat Yourself) principle. (Severity: Medium)
**Suggestion**: Refactor these methods into a single method that takes parameters for the token and service type, reducing redundancy.